### PR TITLE
DOC: use bibtex in plot_sleep

### LINKED ISSUE #8942: use bibtex in plot_sleep.py
convert references in `tutorials/sample-datasets/plot_sleep.py` to use footcite / footbibliography

### PR DESCRIPTION
**Pull request by HanBnrd**
#### Reference issue
Closes #8942.

#### What does this implement/fix?
Added bibtex referencing format to plot_sleep.

### CODE REVIEW

**Comment by HanBnrd** [comment]
Should I create a changefile entry for this kind of bibtex modification?

**Comment by drammock** [comment]
changelog entry is not needed for small documentation tweaks usually.

File: tutorials/sample-datasets/plot_sleep.py [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
+          citing this work.
 
 This tutorial explains how to perform a toy polysomnography analysis that
 answers the following question:
 
-.. important:: Given two subjects from the Sleep Physionet dataset [1]_ [2]_,
-               namely *Alice* and *Bob*, how well can we predict the sleep
-               stages of *Bob* from *Alice's* data?
+.. important:: Given two subjects from the Sleep Physionet dataset
+               :footcite:`KempEtAl2000` :footcite:`GoldbergerEtAl2000`, namely
```
Comment (drammock) [inline tutorials/sample-datasets/plot_sleep.py:16]:
```suggestion
               :footcite:`KempEtAl2000,GoldbergerEtAl2000`, namely
```

File: tutorials/sample-datasets/plot_sleep.py [hunk 2]
```diff
@@ -53,7 +55,8 @@
 #
 # MNE-Python provides us with
 # :func:`mne.datasets.sleep_physionet.age.fetch_data` to conveniently download
-# data from the Sleep Physionet dataset [1]_ [2]_.
+# data from the Sleep Physionet dataset :footcite:`KempEtAl2000`
+# :footcite:`GoldbergerEtAl2000`.
```
Comment (drammock) [inline tutorials/sample-datasets/plot_sleep.py:59]:
```suggestion
# data from the Sleep Physionet dataset
:footcite:`KempEtAl2000,GoldbergerEtAl2000`.
```

**Review by drammock — APPROVED** [review]
FYI, you can stack multiple references within one `footcite` by separating with commas

**Review by hoechenberger — APPROVED** [review]

**Comment by drammock** [comment]
haha, oops, I broke your PR!   I think maybe you can't start a line with `:footcite:`

File: tutorials/sample-datasets/plot_sleep.py [hunk 3]
```diff
@@ -53,7 +55,8 @@
 #
 # MNE-Python provides us with
 # :func:`mne.datasets.sleep_physionet.age.fetch_data` to conveniently download
-# data from the Sleep Physionet dataset [1]_ [2]_.
+# data from the Sleep Physionet dataset
+:footcite:`KempEtAl2000,GoldbergerEtAl2000`.
```
Comment (hoechenberger) [inline tutorials/sample-datasets/plot_sleep.py:59]:
```suggestion
# data from the Sleep Physionet dataset
# :footcite:`KempEtAl2000,GoldbergerEtAl2000`.
```

**Review by hoechenberger — APPROVED** [review]

**Review by larsoner — COMMENTED** [review]
On `main` I see:
```
$ git grep Goldbe
doc/overview/datasets_index.rst:available at PhysioNet :footcite:`GoldbergerEtAl2000`. The dataset contains
doc/overview/datasets_index.rst:data please cite :footcite:`KempEtAl2000` and :footcite:`GoldbergerEtAl2000`.
doc/references.bib:@article{GoldbergerEtAl2000,
doc/references.bib:  author = {Goldberger, Ary L. and Amaral, Luis A. N. and Glass, Leon and Hausdorff, Jeffrey M. and Ivanov, Plamen Ch. and Mark, Roger G. and Mietus, Joseph E. and Moody, George B. and Peng, Chung-Kang and Stanley, H. Eugene},
examples/decoding/plot_decoding_csp_eeg.py::footcite:`GoldbergerEtAl2000`.
mne/datasets/eegbci/eegbci.py:    .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
mne/datasets/eegbci/eegbci.py:    .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
mne/datasets/sleep_physionet/_utils.py:    .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
mne/datasets/sleep_physionet/age.py:    .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
mne/datasets/sleep_physionet/temazepam.py:    .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
tutorials/preprocessing/plot_40_artifact_correction_ica.py:# .. [6] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh, Mark RG,
tutorials/sample-datasets/plot_sleep.py:# .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
```
Do you want to try replacing the others, too? If not I'm happy to merge this as-is and we can fix the others in another PR.

**Comment by HanBnrd** [comment]
Hi @larsoner, I think they've been left to separate issues for other participants of the code sprint to have easy first PRs. But I can do the other files if you want, let me know :)

**Comment by larsoner** [comment]
Ahh makes sense, in it goes. Thanks @HanBnrd !

**Pull request merged**